src/flood_model/bg_flood_model.py

```diff
@@ -301,17 +306,17 @@ def prepare_bg_flood_model_inputs(
         The file path of the Hydrologically conditioned DEM (Hydro DEM) for the specified catchment area.
     resolution : Union[int, float]
         The grid resolution in meters for metric grids, representing the size of each grid cell.
-    output_timestep : Union[int, float], optional
+    output_timestep : Union[int, float] = 0
         Time step between model outputs in seconds. Default value is 0.0 (no output generated).
-    end_time : Union[int, float], optional
+    end_time : Union[int, float] = 0
         Time in seconds when the model stops. Default value is 0.0 (model initializes but does not run).
```

Comment:
Remove the default value from here as well. Since we should not provide defaults that will make things more confusing.
```suggestion
    end_time : Union[int, float]
        Time in seconds when the model stops. If the value is set to 0 then the model initializes but does not run.
```